### PR TITLE
Updated JavaFormatSourcesCommand.

### PR DESCRIPTION
**Pull request by D9110**

### CODE REVIEW

**Comment by gastaldi** [comment]
The formatter profile name can also be stored in the user Configuration and must be checked

**Comment by D9110** [comment]
@gastaldi the formatter profile name should be checked against what ?

**Comment by gastaldi** [comment]
FormatterProfileReader.getSettingsFor(String proileName)

**Comment by D9110** [comment]
@gastaldi Okay I'll refactor the code.
Well don't you think it will be good to add another input field(for all custom format XML's) - the name of the profile.
Or..a function in FormatterProfileReader class that returns the name element of the xml profile passed to it.

**Comment by gastaldi** [comment]
Yes, having another input for the profile name is a good idea. 

As for the names in a profile XML, use FormatterProfileReader.getNames()

**Comment by D9110** [comment]
@gastaldi  I meant a function for get the name of a "particular" xml I pass to it.

**Comment by gastaldi** [comment]
UISelectOne<String> profileName and make it enabled only if the profile XML path is provided

**Comment by gastaldi** [comment]
Not sure what you mean, can you give me an example?

**Comment by D9110** [comment]
When the user goes on registering more and more xmls in the FormatterProfileReader without knowing the name element of the profile . (As in there is not input for the name element currently...when I register my format xml to FormatterProfileReader I donot know its name..hence I cannot keep track of the XML profile I registered (in a situation of multiple xml inputs)...I wont be able to use getPropertiesFor(String formatterName) without the knowledge of the name)....other than when I set up XML parsing in my addon code.

**Comment by D9110** [comment]
@gastaldi So when I was updating this command , I was thinking that either we have a profile name input or we have a utility function to return the name element of the given xml path.That would make the code more robust.

**Comment by gastaldi** [comment]
To fetch all the name elements of the given XML path, use FormatterProfileReader.getProfileNames(). This should be returned in the UISelectOne.getValueChoices()

File: dev-tools-java/src/main/java/org/jboss/forge/addon/devtools/java/JavaFormatSourcesCommand.java [hunk 1]
```diff
@@ -99,8 +118,10 @@ public Result execute(UIExecutionContext context) throws Exception
 
       else
       {
-         formatProfile = new Properties();
-         formatProfile.load(formatProfileLocation.getResourceInputStream());
+         String formatterProfilePath = formatProfileLocation.getFullyQualifiedName();
+         FileInputStream fis = new FileInputStream(formatterProfilePath);
```
Comment (gastaldi) [inline dev-tools-java/src/main/java/org/jboss/forge/addon/devtools/java/JavaFormatSourcesCommand.java]: Put in a try-with-resources block

**Comment by gastaldi** [comment]
Hum, there is an issue here. JavaResourceImpl.setContents() will always use the formatter in the user configuration. To specify a different formatter, we need to add a new setContents(InputStream, String formatterPath, String formatterName) method to the JavaResource interface and use that in this command

**Comment by gastaldi** [comment]
A better signature would be: 
    JavaResource.setContents(InputStream, Resource<?> formatterResource, String formatterName)

**Pull request closed without merging**